File: server_integration_tests/servers/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/TeamServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.TeamService;
import dev.graphlink.javaserver.generated.types.Team;

import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

import java.util.List;

@Service
public class TeamServiceImpl implements TeamService {

    static final List<Team> TEAMS = List.of(Team.builder().id("1").name("Core").build());

    @Override
    public Flux<Team> listTeams() {
        return Flux.fromIterable(TEAMS);
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/TeamServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.services.TeamService
import dev.graphlink.kotlinserverblocking.generated.types.Team
import org.springframework.stereotype.Service

@Service
class TeamServiceImpl : TeamService {
    override fun listTeams(): List<Team> = listOf(Team(id = "1", name = "Core"))
}
